cos-fleetshard-operator-debezium/src/main/java/org/bf2/cos/fleetshard/operator/debezium/model/AbstractApicurioConverter.java

```diff
@@ -0,0 +1,33 @@
+package org.bf2.cos.fleetshard.operator.debezium.model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.bf2.cos.fleetshard.api.ManagedConnector;
+import org.bf2.cos.fleetshard.api.SchemaRegistrySpec;
+import org.bf2.cos.fleetshard.api.ServiceAccountSpec;
+
+public abstract class AbstractApicurioConverter implements KafkaConnectConverter {
+
+    public static final String APICURIO_AUTH_SERVICE_URL = "https://identity.api.openshift.com/auth";
+
+    @Override
+    public Map<String, String> getAdditionalConfig(ManagedConnector config, ServiceAccountSpec serviceAccountSpec) {
+        Map<String, String> additionalConfig = new HashMap<>();
+        additionalConfig.put("apicurio.auth.service.url", APICURIO_AUTH_SERVICE_URL);
+        additionalConfig.put("apicurio.auth.realm", "rhoas");
+
+        SchemaRegistrySpec schemaRegistrySpec = config.getSpec().getDeployment().getSchemaRegistry();
+        if (null == schemaRegistrySpec || null == schemaRegistrySpec.getUrl() || schemaRegistrySpec.getUrl().isBlank()) {
+            throw new RuntimeException("Can't create a schema-based connector without providing a valid 'schema_registry'");
+        }
+        String schemaRegistryURL = schemaRegistrySpec.getUrl();
+
+        additionalConfig.put("apicurio.registry.url", schemaRegistryURL);
+        additionalConfig.put("apicurio.auth.client.id", serviceAccountSpec.getClientId());
+        additionalConfig.put("apicurio.auth.client.secret", serviceAccountSpec.getClientSecret());
```

Comment:
I think the client secret should be loaded using a file protocol or something like https://github.com/bf2fc6cc711aee1a0c2a/cos-fleetshard/blob/b6402e0b5b62a5eee02e83cde41328794d490e2e/cos-fleetshard-operator-debezium/src/main/java/org/bf2/cos/fleetshard/operator/debezium/DebeziumOperandSupport.java#L96-L101

Reply:
@lburgazzoli  are you sure? This goes to the Kafka Connect (not connector!) config, which should be invisible to the user. External config should maybe go to objects that needs to go to the connector config?